Civi/Selfservice/ActionProvider/Action/SendLink.php

```diff
@@ -0,0 +1,106 @@
+| Author: B. Endres (endres@systopia.de)                      |
++-------------------------------------------------------------+
+| This program is released as free software under the         |
+| Affero GPL license. You can redistribute it and/or          |
+| modify it under the terms of this license which you         |
+| can read by viewing the included agpl.txt or online         |
+| at www.gnu.org/licenses/agpl.html. Removal of this          |
+| copyright header is strictly prohibited without             |
+| written permission from the original author(s).             |
++-------------------------------------------------------------*/
+
+namespace Civi\Selfservice\ActionProvider\Action;
+
+use CRM_Selfservice_ExtensionUtil as E;
+
+use \Civi\ActionProvider\Action\AbstractAction;
+use \Civi\ActionProvider\Parameter\ParameterBagInterface;
+use \Civi\ActionProvider\Parameter\Specification;
+use \Civi\ActionProvider\Parameter\SpecificationBag;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class SendLink extends AbstractAction implements CompilerPassInterface {
+
+  /**
+   * Register this one action: SelfServiceResolve
+   */
+  public function process(ContainerBuilder $container) {
+    if (!$container->hasDefinition('action_provider')) {
+      return;
+    }
+    $typeFactoryDefinition = $container->getDefinition('action_provider');
+    $typeFactoryDefinition->addMethodCall('addAction', ['Sendlink', 'Civi\Selfservice\ActionProvider\Action\SendLink', E::ts('Send Self-Service Hash to Email-Adress'), [
+        AbstractAction::SINGLE_CONTACT_ACTION_TAG,
+        AbstractAction::DATA_RETRIEVAL_TAG
+    ]]);
+  }
+
+  /**
+   * Returns the specification of the configuration options for the actual action.
+   *
+   * @return SpecificationBag specs
+   */
+  public function getConfigurationSpecification() {
+    return new SpecificationBag([]);
+  }
+
+  /**
+   * Returns the specification of the parameters of the actual action.
+   *
+   * @return SpecificationBag specs
+   */
+  public function getParameterSpecification() {
+    return new SpecificationBag([
+        new Specification('email', 'String', E::ts('Email'), false, null, null, null, false),
+   ]);
+  }
+
+  /**
+   * Returns the specification of the output parameters of this action.
+   *
+   * This function could be overriden by child classes.
+   *
+   * @return SpecificationBag specs
+   */
+  public function getOutputSpecification() {
+    return new SpecificationBag([
+      new Specification('is_error', 'Integer', E::ts('Is Error'), false, null, null, null, false),
+      new Specification('error_message', 'String', E::ts('Error Message'), false, null, null, null, false),
+      new Specification('count', 'Integer', E::ts('Count'), false, null, null, null, false),
+      new Specification('values', 'String', E::ts('Values'), false, null, null, null, false),
+    ]);
```

Comment:
it would be good to return the case, see [here](https://github.com/systopia/de.systopia.selfservice/blob/0.2/api/v3/Selfservice/Sendlink.php#L59-L102).

Reply:
..but you would have to extend the api a bit

Reply:
``count`` and ``values`` doesn't make sense to me, though

Reply:
values contains the message email sent and is a usefull debug-message. i renamed it to message.
I think changes to the api should be made later or by someone else.